scripts/build-layout-scales.js

```diff
@@ -0,0 +1,1271 @@
+        }
+      `);
+    }, '');
+    css += stripIndent('}');
+
+    css += stripIndent(`\n\n@media ${mediaQueries['--l-screen']} {`);
+    css += scales.reduce((result, scale) => {
+      return result += stripIndent(`
+        .pb${scale}-ml {
+          padding-bottom: ${value(scale)} !important;
+        }
+      `);
+    }, '');
+    css += stripIndent('}');
+
+    css += stripIndent(`\n\n@media ${mediaQueries['--xl-screen']} {`);
+    css += scales.reduce((result, scale) => {
+      return result += stripIndent(`
+        .pb${scale}-mxl {
+          padding-bottom: ${value(scale)} !important;
+        }
+      `);
+    }, '');
+    css += stripIndent('}');
+
+    css += stripIndent(`
+    \n/**
+      * Apply padding on the left.
+      *
+      * @group
+      * @example
+      * <div class='pl24 bg-darken10'>pl24</div>
+      * @memberof Padding
+      */`
+    );
+    css += scales.reduce((result, scale) => {
+      return result += stripIndent(`
+        .pl${scale} {
+          padding-left: ${value(scale)} !important;
+        }
+      `);
+    }, '');
+    css += '/** @endgroup */\n';
+
+    css += stripIndent(`\n\n@media ${mediaQueries['--m-screen']} {`);
+    css += scales.reduce((result, scale) => {
+      return result += stripIndent(`
+        .pl${scale}-mm {
+          padding-left: ${value(scale)} !important;
+        }
+      `);
+    }, '');
+    css += stripIndent('}');
+
+    css += stripIndent(`\n\n@media ${mediaQueries['--l-screen']} {`);
+    css += scales.reduce((result, scale) => {
+      return result += stripIndent(`
+        .pl${scale}-ml {
+          padding-left: ${value(scale)} !important;
+        }
+      `);
+    }, '');
+    css += stripIndent('}');
+
+    css += stripIndent(`\n\n@media ${mediaQueries['--xl-screen']} {`);
+    css += scales.reduce((result, scale) => {
+      return result += stripIndent(`
+        .pl${scale}-mxl {
+          padding-left: ${value(scale)} !important;
+        }
+      `);
+    }, '');
+    css += stripIndent('}');
+
+    return css;
+  };
+
+  variantGenerators.width = function (scales) {
+    function value(v) {
+      return v === '-full' ? '100%' : `${v}px`;
+    }
+
+    let css = stripIndent(`
+    /**
+      * Set an element's width.
+      *
+      * @group
+      * @memberof Sizing
+      * @example
+      * <div class='w96 bg-darken10'>w96</div>
+     */`
+    );
+
+    css += scales.reduce((result, scale) => {
+      return result += stripIndent(`
+        .w${scale} {
+          width: ${value(scale)} !important;
+        }
+      `);
+    }, '');
+    css += '/** @endgroup */\n';
+
+    css += stripIndent(`\n\n@media ${mediaQueries['--m-screen']} {`);
+    css += scales.reduce((result, scale) => {
+      return result += stripIndent(`
+        .w${scale}-mm {
+          width: ${value(scale)} !important;
+        }
+      `);
+    }, '');
+    css += stripIndent('}');
+
+    css += stripIndent(`\n\n@media ${mediaQueries['--l-screen']} {`);
```

Comment:
Ok, I'm pretty convinced by this point that there should be a helper function for generating the media query variants, instead of repeating the same pattern so many times.